### PR TITLE
Move database testing utilities into @prairielearn/postgres

### PR DESCRIPTION
**Pull request by nwalters512**
This allows us to reuse our utilities for spinning up and tearing down test databases across packages, including in `@prairielearn/postgres` itself.

Once sproc handling is moved into a shared package, I plan to introduce a package like `@prairielearn/server-test-utils` that encapsulates migrations and sproc creation as well; we can then reuse that across PrairieLearn and PrairieTest (and any other servers we end up with in the future).

### CODE REVIEW

File: packages/postgres/src/test-utils.ts [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+interface DropDatabaseOptions {
+  database?: string;
+  force?: boolean;
+  closePool?: boolean;
+}
+
+async function createDatabase(
+  options: PostgresTestUtilsOptions,
+  createOptions: CreateDatabaseOptions = {}
```
Comment (mwest1066) [inline packages/postgres/src/test-utils.ts:33]: To help me understand the patterns we are using here, could you have kept all the defaults together (and simplified the later code) with something like the following?
```suggestion
  {dropExistingDatabase = true, configurePool = true, ...more options... }: CreateDatabaseOptions
```

File: packages/postgres/src/test-utils.ts [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+  await createOptions.prepare?.(client);
+
+  if (configurePool) {
+    await defaultPool.initAsync(
+      {
+        user: options.user ?? POSTGRES_USER,
+        host: options.host ?? POSTGRES_HOST,
+        database: getDatabaseNameForCurrentMochaWorker(options.database),
+        // TODO: make these configurable?
```
Comment (mwest1066) [inline packages/postgres/src/test-utils.ts:68]: Seems like we may as well do it now, given that it's annoying to change a package and do the whole publish dance if we need to change this in the future?

**Review by mwest1066 — APPROVED** [review]

File: packages/postgres/src/test-utils.ts (hunk 2 above)
Comment (nwalters512) [reply packages/postgres/src/test-utils.ts:68]: Yep, good call - I meant to do this before requesting review but forgot, sorry!

File: packages/postgres/src/test-utils.ts (hunk 1 above)
Comment (nwalters512) [reply packages/postgres/src/test-utils.ts:33]: Yep, that would be equivalent. I'll make that change!

**Review by mwest1066 — APPROVED** [review]

**Pull request merged**